### PR TITLE
Update Dockerfile to pull manifests from different component repo

### PR DESCRIPTION
**Pull request by zdtsw**
## Description
- leave notebook, modelmesh, monitoring as-is from odh-manifests
- the old implmenetation is to get all from odh-manifests, the new one is to only get above listed from odh-manifests, the rest are defined as a list in makefile.
- another difference is the kserve.  the old implementation gets its manifests from odh-manifests which is the "outcome" of kustomize build, the new implementation is to copy over from kserve git repo (which are the source of the odh-manifests part) so no need "base" in the path
- move fetch logic into `get_all_aminfests.sh`
- instruction  of how to fetch manifests has been added into README

ref: https://github.com/opendatahub-io/opendatahub-operator/issues/424 

## How Has This Been Tested?

**UPDATES:**
new local build image: from 14th Sep  quay.io/wenzhou/opendatahub-operator-catalog:v2.9.424

**OLD:**
Screenshot from 2023-08-10 10-02-30
Screenshot from 2023-08-10 10-38-07

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by etirelli — APPROVED** [review]
lgtm

**Comment by zdtsw** [comment]
> /hold
> 
> Till we get update from components and corresponding ADR is merged

should we get this one out first and keep migrating the rest 2 components?
so they can comment out the theirs from dockerfile and test locally

**Comment by zdtsw** [comment]
ref to ADR: https://github.com/opendatahub-io/architecture-decision-records/pull/16/files

File: components/kserve/kserve.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ import (
 
 const (
 	ComponentName          = "kserve"
-	Path                   = deploy.DefaultManifestPath + "/" + ComponentName + "/base"
+	Path                   = deploy.DefaultManifestPath + "/" + ComponentName
```
Comment (VaishnaviHire) [inline components/kserve/kserve.go:19]: Why is this change required?

Comment (zdtsw) [reply components/kserve/kserve.go:19]: https://github.com/opendatahub-io/kserve/tree/master/config does not have a "`base`" folder.

i will need config get_all_manifests.sh to use 
`config/overlays/odh` as path

**Comment by zdtsw** [comment]
/hold

need to get https://github.com/opendatahub-io/notebooks/pull/186 in first

**Comment by zdtsw** [comment]
/unhold

notebook-images has been updated to get manifets from opendatahub-io/notebooks

**Review by VaishnaviHire — APPROVED** [review]
/lgtm
